app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = [Person(person["name"], person["age"]) for person in people]
+
+    for person_dict in people:
+        name = person_dict["name"]
+        spouse_key = "wife" if "wife" in person_dict else "husband"
+        spouse_name = person_dict.get(spouse_key)
+
+        if spouse_name is not None:
+            # Link the spouse attribute to the corresponding Person instance
+            setattr(
+                Person.people[name],
+                spouse_key,
+                Person.people[spouse_name]
```

Comment:
There is a potential issue here if `spouse_name` is not found in `Person.people`. This will raise a `KeyError`. Consider adding a check to ensure `spouse_name` exists in the dictionary before attempting to access it.